client/src/main/java/io/pravega/client/connection/impl/FlowClientConnection.java

```diff
@@ -70,4 +71,10 @@ public void close() {
         }
     }
 
+    @Override
+    public PravegaNodeUri getLocation() { 
+        String[] nodeLocation = connectionName.split(":");
+        return new PravegaNodeUri(nodeLocation[0].trim(), Integer.parseInt(nodeLocation[1].trim()));
```

Comment:
Don't add parsing logic here. We already have utility functions for this in `shared`. If you can't find what you want in `NameUtils` maybe add it there.

Reply:
Incorporated changes